libcontainer/cgroups/fs2/fs2.go

```diff
@@ -53,7 +53,10 @@ func (m *manager) getControllers() error {
 
 	file := filepath.Join(m.dirPath, "cgroup.controllers")
 	data, err := ioutil.ReadFile(file)
-	if err != nil && !m.rootless {
+	if err != nil {
+		if m.rootless && m.config.Path == "" {
+			return nil
```

Comment:
Does it make sense to keep trying to read the file every time the function is called, or should we maybe use `sync.Once()` here?

Reply:
I guess `sync.Once()` can be used, but this issue is not new in this PR

Reply:
correct